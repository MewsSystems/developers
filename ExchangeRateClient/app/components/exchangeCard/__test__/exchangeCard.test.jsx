import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ExchangeCard from '../index';

const pairInfo = {
  first: {
    name: 'ABC',
    code: 'XYZ',
  },
  second: {
    name: 'DEF',
    code: 'UVW',
  },
  rate: 1.12345,
  pairKey: '70c6744c-cba2-5f4c-8a06-0dac0c4e43a1',
};

const trendDirection = 'down';

describe('Exchange Card', () => {
  const component = shallow(
    <ExchangeCard
      key={pairInfo.pairKey}
      pairInfo={pairInfo}
      trendDirection={trendDirection}
    />,
  );

  it('renders 1 component', () => {
    expect(component).toHaveLength(1);
  });

  it('matches snapshot', () => {
    const layout = renderer
      .create(
        <ExchangeCard
          key={pairInfo.pairKey}
          pairInfo={pairInfo}
          trendDirection={trendDirection}
        />,
      )
      .toJSON();
    expect(layout).toMatchSnapshot();
  });
});
