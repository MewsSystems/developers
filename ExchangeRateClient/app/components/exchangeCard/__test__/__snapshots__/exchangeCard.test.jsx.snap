// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange Card matches snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        XYZ
      </div>
      <div>
        ABC
      </div>
    </div>
    <span>
      arrow_right
    </span>
    <div>
      <div>
        UVW
      </div>
      <div>
        DEF
      </div>
    </div>
  </div>
  <div>
    <div>
      1.12345
    </div>
    <div>
      <span>
        trending_down
      </span>
    </div>
  </div>
</div>
`;
